flink-table-store-core/src/main/java/org/apache/flink/table/store/file/utils/SnapshotManager.java

```diff
@@ -214,13 +217,22 @@ public Optional<Snapshot> latestSnapshotOfUser(String user) {
     public Long readHint(String fileName) {
         Path snapshotDir = snapshotDirectory();
         Path path = new Path(snapshotDir, fileName);
+        int retryNumber = 0;
         try {
-            if (path.getFileSystem().exists(path)) {
-                return Long.parseLong(FileUtils.readFileUtf8(path));
+            while (retryNumber++ < READ_HINT_RETRY_NUM) {
+                if (path.getFileSystem().exists(path)) {
+                    return Long.parseLong(FileUtils.readFileUtf8(path));
```

Comment:
We need to catch exception and ignore exception here.

Reply:
Done.

Reply:
```
int retryNumber = 0;
while (retryNumber++ < READ_HINT_RETRY_NUM) {
        try {
             return Long.parseLong(FileUtils.readFileUtf8(path));
        } catch (Exception ignored) {
        }
       TimeUnit.MILLISECONDS.sleep(READ_HINT_RETRY_INTERVAL);
}
```

Reply:
OK. I've added a fix-up commit. Currently only exception of "Long.parseLong(FileUtils.readFileUtf8(path))" is catched, and others are thrown to invoker.

Reply:
Maybe we can just remove `if (path.getFileSystem().exists(path)) {`, we already have catch the exception for `FileUtils.readFileUtf8(path)`.

Reply:
Done.